flytekit/core/type_engine.py

```diff
@@ -42,22 +44,21 @@
 from flytekit.models import types as _type_models
 from flytekit.models.annotation import TypeAnnotation as TypeAnnotationModel
 from flytekit.models.core import types as _core_types
-from flytekit.models.literals import (
-    Literal,
-    LiteralCollection,
-    LiteralMap,
-    Primitive,
-    Scalar,
-    Union,
-    Void,
-)
+from flytekit.models.literals import Binary, Literal, LiteralCollection, LiteralMap, Primitive, Scalar, Union, Void
 from flytekit.models.types import LiteralType, SimpleType, TypeStructure, UnionType
 
 T = typing.TypeVar("T")
 DEFINITIONS = "definitions"
 TITLE = "title"
 
 
+# In Mashumaro, the default encoder uses strict_map_key=False, while the default decoder uses strict_map_key=True.
```

Comment:
this function doesn't have anything to do with Mashumaro right?  why mention mashumaro in the comments?

Reply:
This function will be put into `_default_flytekit_decoder`.
For example, if we access `dataclasss.dict_int_str` in a workflow, then we will use `from_binary_idl` here to turn the Binary IDL object to `Dict[int, str]`.

Note: `dict_int_str` is `Dict[int, str]`.